Yatima/FromLean.lean

```diff
@@ -159,50 +169,58 @@ mutual
     | none   => throw s!"'{nam}' not found in '{lvls}'"
   | .mvar .. => throw "Unfilled level metavariable"
 
+mutual
+
   partial def toYatimaRecursorRule
     (ctorCid : ConstCid) (rules : Lean.RecursorRule) :
-      EnvM RecursorRule := do
+      ToYatimaM RecursorRule := do
     let rhs ← toYatimaExpr [] rules.rhs
     let rhsCid ← exprToCid rhs
     addToEnv $ .expr_cache rhsCid rhs
     return ⟨ctorCid, rules.nfields, rhsCid⟩
 
   partial def toYatimaExpr (levelParams : List Lean.Name) :
-      Lean.Expr → EnvM Expr
+      Lean.Expr → ToYatimaM Expr
     | .bvar idx _ => return .var "" idx
```

Comment:
I'm not sure about this, but now since we're bookkeeping our own binder count, directly using `idx` might create some duplicated indices where an inductive and variable that's nested one level above/below will clash.

Reply:
Hm, do you have an idea for a workaround?

Reply:
No, I can't think of/conceptually process a complicated binding situation where this would fail yet (that's why I asked about the printer lol)

Reply:
There's no problem with using the `.bvar idx`, as long as you make sure that when you substitute a const for a free var, you do `.var nam nBinders`. The variables won't conflict or duplicate

Reply:
Also, we can add the names of bound variables to the resulting expression instead of`.var ""` by tracking the binding context as a list of names and looking them up via index